common/OpTestSSH.py

```diff
@@ -0,0 +1,180 @@
+from OpTestConstants import OpTestConstants as BMC_CONST
+import OpTestSystem
+try:
+    from common import OPexpect
+except ImportError:
+    import OPexpect
+
+class ConsoleState():
+    DISCONNECTED = 0
+    CONNECTED = 1
+
+def set_system_to_UNKNOWN(system):
+    s = system.get_state()
+    system.set_state(OpTestSystem.OpSystemState.UNKNOWN)
+    return s
+
+class OpTestSSH():
+    def __init__(self, host, username, password, logfile=sys.stdout, port=22,
+            prompt=None):
+        self.state = ConsoleState.DISCONNECTED
+        self.host = host
+        self.username = username
+        self.password = password
+        self.port = port
+        self.logfile = logfile
+        self.prompt = prompt
+        self.system = None
+
+    def set_system(self, system):
+        self.system = system
+
+    def terminate(self):
+        if self.state == ConsoleState.CONNECTED:
+            self.console.terminate()
+            self.state = ConsoleState.DISCONNECTED
+
+    def close(self):
+        if self.state == ConsoleState.DISCONNECTED:
+            return
+        try:
+            self.console.send("\r")
+            self.console.send('~.')
+            self.console.expect(pexpect.EOF)
+            self.console.close()
+        except pexpect.ExceptionPexpect:
+            raise "SSH Console: failed to close ssh console"
+        self.console.terminate()
+        self.state = ConsoleState.DISCONNECTED
+
+    def connect(self):
+        if self.state == ConsoleState.CONNECTED:
+            self.console.terminate()
+            self.state = ConsoleState.DISCONNECTED
+
+        cmd = ("sshpass -p %s " % (self.password)
+               + " ssh -q"
+               + " -p %s" % str(self.port)
+               + " -l %s %s" % (self.username, self.host)
+               + " -o'RSAAuthentication=no' -o 'PubkeyAuthentication=no'"
+               + " -o 'UserKnownHostsFile=/dev/null' "
+               + " -o 'StrictHostKeyChecking=no'"
+           )
+
+        print cmd
+        consoleChild = OPexpect.spawn(cmd,logfile=self.logfile,
+                failure_callback=set_system_to_UNKNOWN,
+                failure_callback_data=self.system)
+        self.state = ConsoleState.CONNECTED
+        self.console = consoleChild
+        # Users expecting "Host IPMI" will reference console.sol so make it available
+        self.sol = self.console
+        return consoleChild
+
+    def get_console(self):
+        if self.state == ConsoleState.DISCONNECTED:
+            self.connect()
+
+        count = 0
+        while (not self.console.isalive()):
+            print '# Reconnecting'
+            if (count > 0):
+                time.sleep(1)
+            self.connect()
+            count += 1
+            if count > 120:
+                raise "SSH: not able to get console"
+
+        return self.console
+
+    def run_command(self, command, timeout=60):
+        console = self.get_console()
+
+        if self.prompt:
+            prompt = self.prompt
```

Comment:
If i use simple ls command it's failing. so please set the unique prompt and run the different tests i mentioned in HostLogin testcase
```/ # ls
ls
SMASH       etc         lost+found  proc        share       usr
bin         lib         mnt         run         sys         var
dev         linuxrc     nv          sbin        tmp         web
/ # echo $?
echo $?
0
/ # ERROR (6.665s)

======================================================================
ERROR [6.665s]: runTest (testcases.OpalMsglog.Skiroot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins_data/sath/pridhivi/sammj_tests/op-test-framework/testcases/OpalMsglog.py", line 35, in setUp
    self.bmc.ssh.run_command("ls")
  File "/home/jenkins_data/sath/pridhivi/sammj_tests/op-test-framework/common/OpTestSSH.py", line 161, in run_command
    exitcode = int(console.before)
ValueError: invalid literal for int() with base 10: '0\r\n/ '

```

Reply:
There's a typo here that meant custom prompts weren't getting set properly - will fix.

Reply:
Sure, Thanks.